src/cs/usfca/edu/histfavcheckout/model/UserRepository.java

```diff
@@ -28,6 +28,12 @@ public int updateCheckoutDetails(@Param("checkouts") boolean checkout, @Param("e
 	@Query("SELECT u FROM User u WHERE u.id.userId = :userId AND u.checkouts = :checkouts")
 	public List<User> findCheckedOutMovies(@Param("userId") int userId, @Param("checkouts") boolean checkout, Pageable pageable);
 
+	@Query("SELECT COUNT(*) FROM User u WHERE (u.checkouts = true) AND (u.id.userId = :userId)")
+	public int getCheckoutCount(@Param("userId") int userId);
+
+	@Query("SELECT u FROM User u where u.id.userId = :userId")
+	public List<User> getUserFavorites(@Param("userId") int userId);
+
```

Comment:
There is an exact same query above `findUserWithUserId()`, please reuse the above one.